main.go

```diff
@@ -126,11 +128,31 @@ func main() {
 		SyncPeriod: custodianSyncPeriod,
 	})
 
+	if err != nil {
+		setupLog.Error(err, "Unable to initialize etcd custodian controller with image vector")
+		os.Exit(1)
+	}
+
 	if err := custodian.SetupWithManager(ctx, mgr, custodianWorkers); err != nil {
 		setupLog.Error(err, "Unable to create controller", "Controller", "Etcd Custodian")
 		os.Exit(1)
 	}
 
+	lc, err := controllers.NewLeaseControllerWithImageVector(mgr, controllersconfig.CompactionConfig{
+		EventsThreshold:        1000000,
+		ActiveDeadlineDuration: "2m",
```

Comment:
You introduced command line flags to configure these values, but here they are set them statically. Either overwrite them here with the configured ones or remove the CMD flags which is probably not intended.